File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        Map<Integer, Integer> mp = new HashMap<>();
        for (int i = 0; i < n; i++){
            mp.put(nums[i], i);
        }
        for (int i = 0; i < n; i++){
            int complement = target - nums[i];
            if(mp.containsKey(complement) && mp.get(complement) != i){
                return new int[] {i, mp.get(complement)};
            }
        }
        return new int[] {};
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        map<int, int> mp;
        for (int i = 0; i < n; i++){
            mp[nums[i]] = i;
        }
        for (int i = 0; i < n; i++){
            int complement = target - nums[i];
            if(mp.contains(complement) && mp[complement] != i){
                return {i, mp[complement]};
            }
        }
        return {};
    }
};